independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -51,11 +51,12 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
         String name = null;
 
         for (AnnotationInstance annotation : beanDeployment.getAnnotations(beanClass)) {
-            if (beanDeployment.getQualifier(annotation.name()) != null) {
+            for (AnnotationInstance qualifierAnnotation : beanDeployment.getQualifierExtractor(annotation.name())
+                    .apply(annotation)) {
                 // Qualifiers
-                qualifiers.add(annotation);
-                if (DotNames.NAMED.equals(annotation.name())) {
-                    AnnotationValue nameValue = annotation.value();
+                qualifiers.add(qualifierAnnotation);
+                if (DotNames.NAMED.equals(qualifierAnnotation.name())) {
```

Comment:
I don't think we need to run this for every annotation. I'd place this `if`  block outside the loop.